resources/schema.py

```diff
@@ -500,17 +500,22 @@ def mutate_and_get_payload(cls, root, info, **input):
 
         width = input.pop("width", None)
         length = input.pop("length", None)
-        depth = input.pop("depth", None)
+        depth_in_input = "depth" in input
         mooring_type = input.pop("mooring_type", None)
+        berth_type = None
 
-        if any([width, length, mooring_type]):
+        if any([width, length, depth_in_input, mooring_type]):
+            depth = input.get("depth", None)
             old_berth_type = berth.berth_type
             berth_type, created = BerthType.objects.get_or_create(
                 width=width or old_berth_type.width,
                 length=length or old_berth_type.length,
-                depth=depth or old_berth_type.depth,
+                # Checking directly if it's in the input keys because it allows null values
+                depth=depth if depth_in_input else old_berth_type.depth,
```

Comment:
I would probably oneline this:
```python
depth=input["depth"] if depth_in_input else old_berth_type.depth,
```
to keep all the logic in one place for better readability

Reply:
How about this?
```python
depth=input.pop("depth", None)
if depth_in_input
else old_berth_type.depth,
```

Reply:
yeah, sure, if you need to pop it.

Reply:
could be just `input.pop("depth")` since we have a check anyway

Reply:
It has to be removed from the input, since the `input` object is passed to update the `berth`, and it might cause trouble, so popping would be the best option.

Reply:
aight 👍